src/CloseCircle.js

```diff
@@ -4,33 +4,35 @@ import React, {PropTypes} from 'react';
 const RECT_WIDTH = 1.5;
 const MARGIN = 8;
 
-const CloseCircle = (props) => {
-  const {
-    diameter,
-  } = props;
+class CloseCircle extends React.PureComponent {
```

Comment:
What's the benefit of using PureComponent (over a function? I mean I guess are functional components not treated as implicitly pure?) and which version of react was this first introduced in?